crates/batch-submitter/src/builder.rs

```diff
@@ -0,0 +1,266 @@
+    tx.output.consensus_encode(&mut encoder).unwrap();
+    tx.lock_time.consensus_encode(&mut encoder).unwrap();
+
+    encoder
+}
+
+fn fund_tx(ctx: &BuilderContext, tx: Transaction) -> Result<Transaction, BatchSubmitterError> {
+    let tx_raw = encode_tx_non_segwit(tx);
+    let funded_tx = ctx
+        .rpc_client
+        .fund_raw_transaction(
+            &tx_raw,
+            Some(&json::FundRawTransactionOptions {
+                fee_rate: ctx.fee_rate.fee_vb(1000), // convert to sat/kvB
+                ..Default::default()
+            }),
+            None,
+        )?
+        .transaction()
+        .unwrap();
+
+    Ok(funded_tx)
+}
+
+fn generate_key_pair() -> Result<UntweakedKeypair, anyhow::Error> {
+    let mut rand_bytes = [0; 32];
+    OsRng.fill_bytes(&mut rand_bytes);
+    Ok(UntweakedKeypair::from_seckey_slice(SECP256K1, &rand_bytes)?)
+}
+
+fn build_reveal_script(
+    taproot_public_key: &XOnlyPublicKey,
+    payload: &[u8],
+) -> Result<ScriptBuf, anyhow::Error> {
+    let mut script_builder = script::Builder::new()
+        .push_x_only_key(taproot_public_key)
+        .push_opcode(bitcoin::opcodes::all::OP_CHECKSIG)
+        .push_opcode(bitcoin::opcodes::all::OP_IF);
+
+    const MAX_PUSH_SIZE: usize = 520;
+    for chunk in payload.chunks(MAX_PUSH_SIZE) {
+        script_builder = script_builder.push_slice(script::PushBytesBuf::try_from(chunk.to_vec())?);
+    }
+    script_builder = script_builder.push_opcode(bitcoin::opcodes::all::OP_ENDIF);
+
+    Ok(script_builder.into_script())
+}
+
+// Estimate the required input value for reveal_tx
+fn calculate_reveal_input_value(
+    amount: Amount,
+    fee_rate: FeeRate,
+    recipient: Address,
+    reveal_script: &script::ScriptBuf,
+    control_block: &ControlBlock,
+) -> Amount {
+    let tx = Transaction {
+        version: Version::TWO,
+        input: vec![TxIn {
+            previous_output: OutPoint {
+                txid: Txid::from_str(
+                    "0000000000000000000000000000000000000000000000000000000000000000",
+                )
+                .unwrap(),
+                vout: 0,
+            },
+            script_sig: script::Builder::new().into_script(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::from_slice(&[
+                Signature::from_slice(&[0; SCHNORR_SIGNATURE_SIZE])
+                    .unwrap()
+                    .as_ref()
+                    .to_vec(),
+                reveal_script.to_bytes(),
+                control_block.serialize(),
+            ]),
+        }],
+        output: vec![TxOut {
+            script_pubkey: recipient.script_pubkey(),
+            value: amount,
+        }],
+        lock_time: LockTime::ZERO,
+    };
+    fee_rate.fee_vb(tx.vsize() as u64).unwrap() + amount
+}
+
+fn build_unfunded_commit_tx(
+    recipient: Address,
+    output_value: Amount,
+) -> Result<Transaction, BatchSubmitterError> {
```

Comment:
Please could you redo the Result type so that you'll have a Result alias type, look in other crate and/or in the Project_structure.md